core/src/main/java/org/mule/config/DefaultMuleConfiguration.java

```diff
@@ -424,6 +427,26 @@ protected void validateXML() throws FatalException
         }
     }
 
+    /**
+     * Register a callback to be called when the logging configuration is changed.
+     * 
+     * @param listener the callback to execute
+     */
+    public static void registerLogConfigChangeListener(PropertyChangeListener listener)
```

Comment:
MuleConfiguration shouldn't expose behaviour, rather just expose configuration values.  See what others think maybe, but i'd prefer these methods were in a LogUtils than here.